homeassistant/components/growatt_server/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for growatt server integration."""
+import voluptuous as vol
+
+import growattServer
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_USERNAME, CONF_PASSWORD
+from .const import DOMAIN, CONF_PLANT_ID, DEFAULT_NAME
+
+
+class GrowattServerConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow class."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialise growatt server flow."""
+        self.user_input = {}
+        self.api = growattServer.GrowattApi()
+
+    async def async_step_plant(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        user_input = {**self.user_input, **user_input}
+
+        if CONF_PLANT_ID not in user_input:
+            login_response = self.api.login(
```

Comment:
Why would we log in again ? We can only get here if we are logged in, no ?  (also I/O inside event loop)

Reply:
This time we want to get the User Id from the login response.
Before we just check if we can log in.

I could make it save the user id in `self` and retrieve this later on. so it doen't have to log in again.

Reply:
Solved this by saving the user id in `self.user_id`